File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/response/Response.java
package lunalauf.rms.utilities.network.communication.message.response;

import com.google.gson.annotations.SerializedName;
import lunalauf.rms.utilities.network.communication.message.Message;
import lunalauf.rms.utilities.network.communication.message.type.MessageType;
import lunalauf.rms.utilities.network.communication.message.type.ResponseType;

public class Response extends Message {

    @SerializedName("res-type")
    public final String responseType;

    public Response(ResponseType responseType) {
        super(MessageType.RESPONSE);
        this.responseType = responseType.name();
    }

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/response/Response.kt
package lunalauf.rms.utilities.network.communication.message.response

import com.google.gson.annotations.SerializedName
import lunalauf.rms.utilities.network.communication.message.Message
import lunalauf.rms.utilities.network.communication.message.type.MessageType
import lunalauf.rms.utilities.network.communication.message.type.ResponseType

open class Response(responseType: ResponseType) : Message(MessageType.RESPONSE) {
    @SerializedName("res-type")
    val responseType: String = responseType.name

    override fun toString(): String {
        return super.toString()
            .replace(")", ", type: $responseType)")
    }
}
